cloud/terraform/terraform.py

```diff
@@ -45,39 +44,45 @@ def __new_aws_provider(self, region):
             'profile': self.aws_profile,
         }
 
-        return new_provider
+        self.providers['provider'][self.cloud].append(new_provider)
+
+    def add_instance_conf(self, instance):
+        if self.cloud == 'aws':
+            self.__new_aws_instance(instance)
+
+    def __new_aws_instance(self, instance):
+        if not instance['instance_type']:
+            instance['instance_type'] = 't2.micro'
+
+        name = instance['name'].replace('.', '-')
 
-    def add_instance_conf(self, region, image_id, name, instance_type=''):
+        # TODO: agree on a name for 'region' or 'location' across all cloud so we can move
```

Comment:
I think we can keep it like this here, since at this point we are handling AWS-related instances data.

If "region" as such doesn't exist for other clouds, we will handle that in the `__new_<cloud>_instance()` methods, like we do here.